### PR TITLE
Prevent traceback at exit

### PR DESCRIPTION
**Pull request by lriesebos**
Just catching the `KeyboardInterrupt` to prevent a traceback print at exit.

Signed-off-by: Leon Riesebos <leon.riesebos@duke.edu>

### CODE REVIEW

**Comment by sbourdeauducq** [comment]
What happens to the controllers then? Can we send them graceful termination requests?

**Comment by lriesebos** [comment]
I would say the proposed change doesn't change much of the exit behavior and controllers are still terminated gracefully.

1. [the keyboard interrupt is raised here, the line I wrapped in a try/except statement](https://github.com/m-labs/artiq-comtools/blob/master/artiq_comtools/artiq_ctlmgr.py#L84). my change will cause that exception to be caught after which the program reached the end of its `main()` function. as a result, the program will exit.
2. now `atexit` will kick in. [you registered `rpc.stop` as the last exit coroutine](https://github.com/m-labs/artiq-comtools/blob/master/artiq_comtools/artiq_ctlmgr.py#L82). the last registered exit coroutine will run first as specified [in the documentation](https://docs.python.org/3.8/library/atexit.html). the rpc server will stop.
3. now [the control manager will be stopped due to this exit coroutine registration](https://github.com/m-labs/artiq-comtools/blob/master/artiq_comtools/artiq_ctlmgr.py#L73). that will [cancel the controller manager task](https://github.com/m-labs/sipyco/blob/master/sipyco/asyncio_tools.py#L24) and raise the `CancelledError` in the coroutine.
4. the controller manager task gets the `CancelledError` and [shuts down controllers gracefully in the finally statement](https://github.com/m-labs/artiq-comtools/blob/af10c3e35efbab515fd5c1c504eecf1bd88f5392/artiq_comtools/ctlmgr.py#L264)

So basically the controller manager still closes all controllers, the same as it would if the exception was not caught. Though now no traceback is printed due to an uncaught `KeyboardInterrupt` exception. This behavior is confirmed both on Linux and Windows in our lab.

If you still have doubts, then you do not have to re-merge the change. It is not that important.

**Comment by sbourdeauducq** [comment]
Thinking more about it, I'm not a huge fan of this traceback suppression. It legitimizes ``KeyboardInterrupt`` as a way to terminate the programs, when it is in fact fundamentally broken (as I explained in the other issue) and simply works most of the time in practice.
If we do want clean Ctrl-C support, someone needs to go down the rabbit hole of ``signalfd`` and friends, plus multi-platform complications.
